src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  function add(a) {
+    if (a === undefined) {
+      const lastResult = result;
+
+      result = 0;
+
+      return lastResult;
+    } else {
+      result += a;
+    }
+
```

Comment:
no needs to add `else` since you've used `return` at `if`
```suggestion
    }
    
    result += a;

```